EpiAware/src/latent-processes.jl

```diff
@@ -5,18 +5,14 @@ function default_rw_priors()
     )
 end
 
-@model function random_walk(
-    n,
-    ϵ_t = missing,
-    ::Type{T} = Float64;
-    latent_process_priors = default_rw_priors(),
-) where {T<:AbstractFloat}
-    rw = Vector{T}(undef, n)
+@model function random_walk(n; latent_process_priors = default_rw_priors())
     ϵ_t ~ MvNormal(ones(n))
     σ²_RW ~ latent_process_priors.var_RW_dist
     init ~ latent_process_priors.init_rw_value_dist
     σ_RW = sqrt(σ²_RW)
+    rw = Vector{eltype(ϵ_t)}(undef, n)
```

Comment:
can you say a bit more about what motivates this change?

Reply:
The previous `function(...args...) where T` pattern didn't work with `@submodel`. 

Rather than burrow hard into that this pattern is flexible enough to give type information to the compiler in the case that `ϵ_t` values aren't Floats (this can happen in forward mode autodiff where `Dual` type numbers are used).